doc/antora/modules/howto/pages/data-usage-reporting/index.adoc

```diff
@@ -0,0 +1,222 @@
+without impacting normal usage reporting.
+
+For reporting purposes, the adhoc periods within the `data_usage_by_period`
+table can be efficiently aggregated to provide data bucketed into the required
+reporting intervals.
+
+
+== Example setup
+
+Megacorp stores their RADIUS accounting data in a MySQL database using the
+standard FreeRADIUS schema. They want to provide reports of monthly data usage
+for given users.
+
+Firstly, implement the extended schema to create the `data_usage_by_period`
+table to hold the data usage amounts and the `fr_new_data_usage_period` stored
+procedure that updates this table:
+
+    mysql radius < /etc/raddb/mods-config/sql/main/mysql/process-radacct.sql
+
+NOTE: The above command assumes that passwordless login has been configured via
+the user's `~/.my.cnf` file, or otherwise. This command and subsequent commands
+should be adjusted accordingly where this is not the case.
+
+Call the update process manually to create the initial accounting period for
+the historical data usage and start of the next period for future use:
+
+    $ echo “CALL fr_new_data_usage_period()” | mysql radius
+
+After some time, once a few sessions have reported in, call the update process
+again manually to populate the current period:
+
+    $ echo “CALL fr_new_data_usage_period()” | mysql radius
+
+Ensure that data is being correctly populated. The data within the
+`data_usage_by_period` table will look something like this (for a single user):
+
+    mysql> SELECT * FROM data_usage_by_period;
+
+    +----------------------+----------------------+----------+-----------------+------------------+
+    | period_start         | period_end           | username | acctinputoctets | acctoutputoctets |
+    +----------------------+----------------------+----------+-----------------+------------------+
+    ...
+    | 1970-01-01T00:00:00Z | 2020-01-01T12:16:42Z | bob      |      1234234324 |      43523453432 |
+    | 2020-01-01T12:16:43Z | 2020-01-01T12:35:20Z | bob      |            4342 |              434 |
+    | 2020-01-01T12:35:21Z | NULL                 | bob      |           -3456 |             -567 |
+    ...
+    +----------------------+----------------------+----------+-----------------+------------------+
+
+Rows with `period_start` is `1970-01-01` represent the data usage by all sessions
+present in the `radacct` table when the update process was initial ran. Such rows
+can be removed at any time if data for this historic period is not required.
+
+Rows where `period_end` is `NULL` represent the open-ended period for future
+usage.  The `acctinputoctets` and `acctoutputoctets` for these rows will be
+zero or negative. _These rows must not be removed, otherwise data for ongoing
+sessions will be accounted for more than once._
+
+For this example we stated that the required reporting interval is initially
+monthly, however it is easy to postulate a future requirement to increase the
```

Comment:
Postulate is pushing the envelope for those who do not speak angleterreish as a primary language.